events/tests/test_api.py

```diff
@@ -304,10 +303,10 @@ def autouse_db(db):
 """
 
 
-def test_events_query_unauthenticated(api_client):
+def test_events_query_unauthenticated(api_client, event, snapshot):
```

Comment:
I guess the API now returns the same result when doing the query with a user or as unauthenticated user? Instead of having to somewhat similar tests we could pytest.parametrize to run the test with different api clients?

```

@pytest.mark.parametrize("client", [api_client, user_api_client])
def test_events_query(client, event, snapshot):
    ...
```

Not sure if the that actually works, but something you can consider if you want.

Reply:
I don't think you can use fixtures in parametrize like that in plain pytest, you'll need an external package. Anyway, as these are probably needed in multiple tests, I think the most DRY solution for this would be to create a new parametrized fixture for the different api clients, something like:

```
@pytest.fixture(params=["api_client", "user_api_client", "guardian_api_client"])
def non_staff_api_clients(request):
    return request.getfixturevalue(request.param))


def test_x(non_staff_api_clients):
    non_staff_api_clients.execute(...)
```

Reply:
Thanks for the comments. 
We discussed that those get API should be only visible to logged in users so I'm gonna close this PR
But yeah I'll refactor the fixtures soon when it's needed